pipeline/outputs/stackdriver.md

```diff
@@ -78,6 +79,75 @@ This implies that if there is no local_resource_id in the log entry then the tag
     tag_prefix  custom_tag.
 ```
 
+## Resource Labels
+
+Currently, there are four ways which fluent-bit uses to assign fields into the resource/labels section.
+
+1. Resource Labels API
+2. Monitored Resource API
+3. Local Resource Id
+4. Credentials / Config Parameters
+
+If `resource_labels` is correctly configured, then fluent-bit will attempt to populate all resource/labels using the entries specified. Otherwise, fluent-bit will attempt to use the monitored resource API. Similarly, if the monitored resource API cannot be used, then fluent-bit will attempt to populate resource/labels using configuration parameters and/or credentials specific to the resource type. As mentioned in the [Configuration File](#configuration-file) section, fluent-bit will attempt to use or construct a local resource ID for a K8s resource type which does not use the resource labels or monitored resource API.
+
+Note that `project_id` will always be set using the `export_to_project_id` parameter or from credentials and cannot be overridden.
```

Comment:
`project_id` won't be set by `export_to_project_id`. `export_to_project_id` sets the project_id in the logName which controls the gcp project the log will be sent to. But `project_id` will be assigned into the resource labels to indicate the service account (or source gcp project) related to this log.

https://sourcegraph.com/github.com/fluent/fluent-bit/-/blob/plugins/out_stackdriver/stackdriver.c?L2150